src/Storage/SalesforceAuthTokenStorage.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace Drupal\salesforce\Storage;
+
+use Drupal\Core\State\StateInterface;
+use Drupal\salesforce\SalesforceAuthProviderPluginManager;
+use Drupal\salesforce\Token\SalesforceToken;
+use OAuth\Common\Storage\Exception\TokenNotFoundException;
+use OAuth\Common\Token\TokenInterface;
+
+class SalesforceAuthTokenStorage implements SalesforceAuthTokenStorageInterface {
+
+  const TOKEN_STORAGE_PREFIX = "salesforce.auth_tokens";
+  const AUTH_STATE_STORAGE_PREFIX = "salesforce.auth_state";
+  const IDENTITY_STORAGE_PREFIX = "salesforce.auth_identity";
+
+  protected $state;
+
+  /**
+   * @deprecated interim, do not use.
+   */
+  private $client;
+
+  public function __construct(StateInterface $state) {
+    $this->state = $state;
+  }
+
+  /**
+   * @deprecated interim, do not use.
+   * @return string
+   */
+  private function service() {
+    $oauth = SalesforceAuthProviderPluginManager::getAuthConfig();
+    return $oauth->id();
+  }
+
+  /**
+   * @deprecated interim, do not use.
+   */
+  public function updateToken() {
+    $this->storeAccessToken($this->service(),
+      new SalesforceToken(
+        $this->state->get('salesforce.access_token'),
+        $this->state->get('salesforce.refresh_token')));
+    return $this;
+  }
+
+  /**
+   * @deprecated interim, do not use.
+   */
+  public function updateIdentity() {
+    $this->storeIdentity($this->service(), $this->state->get('salesforce.identity'));
+    return $this;
+  }
+
+  protected static function getTokenStorageId($service) {
+    return self::TOKEN_STORAGE_PREFIX . '.' . $service;
+  }
+
+  protected static function getAuthStateStorageId($service) {
+    return self::AUTH_STATE_STORAGE_PREFIX . '.' . $service;
+  }
+
+  protected static function getIdentityStorageId($service) {
+    return self::IDENTITY_STORAGE_PREFIX . '.' . $service;
+  }
+
+  /**
+   *{@inheritdoc}
+   */
+  public function retrieveAccessToken($service) {
+    if ($token = $this->state->get(self::getTokenStorageId($service))) {
+      return $token;
+    }
+    throw new TokenNotFoundException();
+  }
+
+  /**
+   *{@inheritdoc}
+   */
+  public function storeAccessToken($service, TokenInterface $token) {
+    $this->state->set(self::getTokenStorageId($service), $token);
+    return $this;
+  }
+
+  /**
+   *{@inheritdoc}
+   */
+  public function hasAccessToken($service) {
+    try {
+      return !empty($this->retrieveAccessToken($service));
+    }
+    catch (TokenNotFoundException $e) {
+      return FALSE;
+    }
+  }
+
+  /**
+   *{@inheritdoc}
+   */
+
+  public function clearToken($service) {
+    $this->state->delete(self::getTokenStorageId($service));
+    return $this;
+  }
+
+  /**
+   *{@inheritdoc}
+   */
+  public function clearAllTokens() {
+    // noop. We don't do this. Only here to satisfy interface.
```

Comment:
This should probably return it self to since the interface says it returns an instance of TokenStorageInterface